test/unit/mdc-menu/menu.foundation.test.js

```diff
@@ -80,1165 +51,431 @@ test('exports cssClasses', () => {
-  foundation.setRememberSelection(true);
-  handlers.click({target});
-  clock.tick(numbers.SELECTED_TRIGGER_DELAY);
-
-  td.when(mockAdapter.getIndexForEventTarget(target)).thenReturn(expectedIndex - 1);
+  const key = {target: {tagName: 'li'}, preventDefault: td.func('preventDefault')};
+  td.when(mockAdapter.elementContainsClass(key.target, listClasses.LIST_ITEM_CLASS)).thenReturn(true);
+  td.when(mockAdapter.getElementIndex(key.target)).thenReturn(0);
+  td.when(mockAdapter.getParentElement(key.target)).thenReturn(key.target);
+  td.when(mockAdapter.elementContainsClass(key.target, cssClasses.MENU_SELECTION_GROUP)).thenReturn(true);
+  td.when(mockAdapter.getSelectedElementIndex(key.target)).thenReturn(0);
 
-  handlers.click({target});
-  clock.tick(numbers.SELECTED_TRIGGER_DELAY);
-
-  td.verify(mockAdapter.rmClassForOptionAtIndex(expectedIndex, td.matchers.anything()),
-    {times: 1});
+  foundation.handleClick(key);
+  clock.tick(numbers.TRANSITION_CLOSE_DURATION);
 
+  td.verify(mockAdapter.removeClassFromElementAtIndex(0, cssClasses.MENU_SELECTED_LIST_ITEM), {times: 1});
+  td.verify(mockAdapter.addClassToElementAtIndex(0, cssClasses.MENU_SELECTED_LIST_ITEM), {times: 1});
   clock.uninstall();
 });
 
-test('should do nothing when the same item is selected twice in a row', () => {
+test('Click event inside of a selection group with no element selected', () => {
   const {foundation, mockAdapter} = setupTest();
-  const handlers = captureHandlers(mockAdapter, 'registerInteractionHandler');
   const clock = lolex.install();
-  const target = {};
-  const expectedIndex = 2;
-  td.when(mockAdapter.getIndexForEventTarget(target)).thenReturn(expectedIndex);
-  td.when(mockAdapter.getNumberOfItems()).thenReturn(3);
-
-  foundation.init();
-  foundation.setRememberSelection(true);
-  handlers.click({target});
-  clock.tick(numbers.SELECTED_TRIGGER_DELAY);
+  const key = {target: {tagName: 'li'}, preventDefault: td.func('preventDefault')};
+  td.when(mockAdapter.elementContainsClass(key.target, listClasses.LIST_ITEM_CLASS)).thenReturn(true);
+  td.when(mockAdapter.getElementIndex(key.target)).thenReturn(0);
+  td.when(mockAdapter.getParentElement(key.target)).thenReturn(key.target);
+  td.when(mockAdapter.elementContainsClass(key.target, cssClasses.MENU_SELECTION_GROUP)).thenReturn(true);
+  td.when(mockAdapter.getSelectedElementIndex(key.target)).thenReturn(-1);
 
-  td.when(mockAdapter.getIndexForEventTarget(target)).thenReturn(expectedIndex);
-
-  handlers.click({target});
-  clock.tick(numbers.SELECTED_TRIGGER_DELAY);
-
-  td.verify(mockAdapter.rmClassForOptionAtIndex(expectedIndex, td.matchers.anything()),
-    {times: 0});
-  td.verify(mockAdapter.addClassForOptionAtIndex(expectedIndex, td.matchers.anything()),
-    {times: 1});
+  foundation.handleClick(key);
+  clock.tick(numbers.TRANSITION_CLOSE_DURATION);
 
+  td.verify(mockAdapter.removeClassFromElementAtIndex(0, cssClasses.MENU_SELECTED_LIST_ITEM), {times: 0});
```

Comment:
See earlier comment RE 0 -> td.matchers.isA(Number) for these `{times: 0}` verifications